packages/core/node/core-node-server-internal/src/node_service.ts

```diff
@@ -52,13 +60,22 @@ export class NodeService {
     loggingSystem.setGlobalContext({ service: { node: { roles } } });
     this.log.info(`Kibana process configured with roles: [${roles.join(', ')}]`);
 
+    this.roles = NODE_ACCEPTED_ROLES.reduce((acc, curr) => {
+      return { ...acc, [camelCase(curr)]: roles.includes(curr) };
+    }, {} as NodeRoles);
+
     return {
-      roles: NODE_ACCEPTED_ROLES.reduce((acc, curr) => {
-        return { ...acc, [camelCase(curr)]: roles.includes(curr) };
-      }, {} as NodeRoles),
+      roles: this.roles,
     };
   }
 
+  public async start(): Promise<InternalNodeServiceStart> {
+    if (this.roles == null) {
```

Comment:
nit: should we validate this against undefined?

Reply:
`this.roles == null` is the same as `this.roles === null && this.roles === undefined`